idseq_dag/steps/run_alignment_remotely.py

```diff
@@ -404,16 +422,24 @@ def run_chunk(self, part_suffix, input_files, chunk_count, lazy_run):
             if status == "SUCCEEDED":
                 break
             if status == "FAILED":
-                log.log_event("alignment_batch_error", values={"job_id": job_id})
+                log.log_event("alignment_batch_job_failed", values={'job_id': job_id, 'chunk_id': chunk_id, 'alignment_algorithm': self.alignment_algorithm})
                 raise Exception("chunk alignment failed")
             time.sleep(delay)
         else:
-            log.log_event("alignment_batch_error", values={"job_id": job_id})
+            log.log_event("alignment_batch_job_timeout", values={'job_id': job_id, 'chunk_id': chunk_id, 'alignment_algorithm': self.alignment_algorithm})
             raise Exception("chunk timed out but never entered the FAILED state")
 
-        fetch_from_s3(multihit_s3_outfile, multihit_local_outfile, okay_if_missing=True, allow_s3mi=False)
+        log.log_event("alignment_batch_job_succeeded", values={'job_id': job_id, 'chunk_id': chunk_id, 'alignment_algorithm': self.alignment_algorithm})
+
+        for _ in range(60):
```

Comment:
Why 60 times in 1 sec? Is waiting a minute that bad?

Reply:
I recommended this pattern - I don't see why we should not retry as quickly as we can reasonably expect the state to change. Waiting 1 minute multiple times could stall the pipeline and time out the job.

Reply:
I just do not enjoy these kind of polling patterns, specially on jobs that are being parallelized at a higher level. The level of parallelization is not controlled from here. Overall, there seems like there are ways to subscribe to events like these.

Reply:
I agree - we need to move to an event-driven model, where this job will not be trying to consume the results of the other job here (for example, by using AWS Batch job dependencies to launch the alignment jobs and schedule a job that depends on all of them - so it doesn't start until they all succeed - and consumes their results). However, that change is outside the scope of this PR.

In general, unfortunately, there is no way to avoid this pattern in a distributed system. This section of logic is subject to S3 eventual consistency, and there is no way to generally specify whether the absence of an S3 key should immediately trigger an error or if should we wait for it to appear (although factoring this out into a more self-explanatory helper function like `wait_for_s3_object()` may make the intent more clear).

Reply:
An even nicer way could be to apply a decorator like `@eventually_consistent` to `fetch_from_s3()`.

Reply:
But i thought that relying on s3 was already a solution to another polling issue. Whoever pushes the s3 object could also push the notification.
Anyway, in the short term i was mostly questioning the frequency...